R/data.table.R

```diff
@@ -2532,7 +2531,7 @@ setnames <- function(x,old,new) {
       if (!is.character(old)) stop("'old' is type ",typeof(old)," but should be integer, double or character")
       if (any(duplicated(old))) stop("Some duplicates exist in 'old': ", paste(old[duplicated(old)],collapse=","))
       i = chmatch(old,names(x))
-      if (anyNA(i)) stop("Items of 'old' not found in column names: ",paste(old[is.na(i)],collapse=","))
+      if (anyNA(i)){ if (skip_absent == TRUE){ w <- old %chin% names(x); old = old[w]; new = new[w]; i = i[w] } else { stop("Items of 'old' not found in column names: ",paste(old[is.na(i)],collapse=",")) } }
```

Comment:
please split the branch logic onto new lines, it's a bit hard to read at the moment